src/api.js

```diff
@@ -281,6 +281,48 @@ api.declare({
 });
 
 api.declare({
+  name: 'badge',
+  title: 'Latest Build Status Badge',
+  description: [
+    'Checks the status of the latest build of a given branch ',
+    'and returns corresponding badge image.',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'get',
+  route: '/badge/:owner/:repo/:branch',
+}, async function(req, res) {
+  // Extract owner, repo and branch from request into variables
+  let {owner, repo, branch} = req.params;
+
+  // Look up the installation ID in Azure. If no such owner in the table, no error thrown
+  let ownerInfo = await this.OwnersDirectory.load({owner}, true);
+
+  // This has nothing to do with user input, so we should be safe
+  let fileConfig = {root : __dirname + '/../assets/'};
+
+  if (ownerInfo) {
+    try {
+      let instGithub = await this.github.getInstallationGithub(ownerInfo.installationId);
+
+      let taskclusterBot = await instGithub.users.getForUser({username: 'taskcluster[bot]'});
+      // Statuses is an array of status objects, where we find the relevant status
+      let statuses = await instGithub.repos.getStatuses({owner, repo, ref: branch});
+      let status = statuses.find(statusObject => statusObject.creator.id === taskclusterBot.id);
+
+      // Then we send a corresponding image. (The path should be absolute)
+      return res.sendFile(status.state + '.svg', fileConfig);
+    } catch (e) {
+      if (e.code < 500) {
```

Comment:
I think in this case we also want to return this image if there's a 400.

The logic here is probably more like

```js
res.sendFile('error.svg', fileConfig);
if (e.code < 500) {
  throw e;
}
```